quarry/data/packets.csv

```diff
@@ -80,7 +80,8 @@ minecraft_version,protocol_version,protocol_mode,packet_direction,packet_name
 1.12.1,338,play,downstream,advancements
 1.12.1,338,play,downstream,entity_effect
 1.12.1,338,play,upstream,teleport_confirm
-1.12.1,338,play,upstream,prepare_crafting_grid
+1.12.1,338,play,upstream,craft_recipe_request
+1.12.1,338,play,downstream,craft_recipe_response
 1.12.1,338,play,upstream,tab_complete

```

Comment:
Getting closer - move the craft_recipe_request line after steer_boat, and craft_recipe_response after open_sign_editor.
Then you're good :)